### PR TITLE
Performance enhancements

### PR DESCRIPTION
**Pull request by alanj853**
This PR is for a performance enhancement with regards to FTP. From my tests, I saw a 3-4% increase in CPU availability. It is worth noting that the FTP server was idling at this time with no clients connecting, and the NMC used was a chassis.

**The Fprof Profiling Test:**
1. Set a timer for exactly 5 mins as soon as the NMC reboots. Do not touch the NMC or log into it for 5 mins.
2. After 5 mins, start the profiling for exactly 20 seconds.

**The Performance Test:**
1. Set a timer for exactly 10 mins as soon as the NMC reboots. Do not touch the NMC or log into it for 10 mins.
2. After 10 mins, extract the performance_info.csv file.

I ran the Performance Test twice, (twice with this fix and twice without), and ran the Fprof Profiling Test once (once with this fix and once without). 

I have attached the results for anyone who wants to see.
[ftp_performance_testing.zip](https://github.com/se-apc/ftp/files/4421914/ftp_performance_testing.zip)

### CODE REVIEW

File: src/connection_supervisor.erl [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+-export([start_link/1]).
+
+%% gen_server callbacks
+-export([init/1, handle_info/2, handle_call/3, handle_cast/2, terminate/2]).
+
+start_link([Args]) ->
+    gen_server:start_link(?MODULE, Args, []).
+
+init(InitialState) ->
```
Comment (michaelkschmidt) [inline src/connection_supervisor.erl:13]: Does this need a `process_flag(trap_exit, true)` call?

File: src/connection_supervisor.erl [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+handle_cast(_Msg, State) ->
+    {noreply, State}.
+
+handle_info({new_connection, Acceptor, Socket}, InitialState) ->
+    Worker = proc_lib:spawn_link(bifrost, establish_control_connection, [Socket, InitialState]),
+            Acceptor ! {ack, Worker},
+    {noreply, InitialState};
+
+handle_info({'EXIT', _Pid, normal}, State) ->
+    {stop, normal, State};
```
Comment (michaelkschmidt) [inline src/connection_supervisor.erl:29]: The original code just continued, it did not stop here

File: src/connection_supervisor.erl [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+handle_info({'EXIT', _Pid, shutdown}, State) ->
+    {stop, shutdown, State};
```
Comment (michaelkschmidt) [inline src/connection_supervisor.erl:32]: Same here--it must continue

File: src/connection_supervisor.erl [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+handle_info({'EXIT', Pid, Info}, State) ->
+    error_logger:error_msg("Control connection ~p crashed: ~p~n", [Pid, Info]),
+    {stop, bad_exit, State};
```
Comment (michaelkschmidt) [inline src/connection_supervisor.erl:36]: And here.  All three need to return {noreply, State}

**Comment by alanj853** [comment]
Latest commits show a CPU decrease of 1-3%.
[performance_info_test.zip](https://github.com/se-apc/ftp/files/4428272/performance_info_test.zip)

**Comment by alanj853** [comment]
Final Test Results: 
[06042020_tests.zip](https://github.com/se-apc/ftp/files/4439016/06042020_tests.zip)
 Summary:
CPU Usage: -1.83%
MEM_AVAILIBLE: +2.4MB
MEM_FREE: -1.1MB (Taken over average, had the most variance in all tests. In some tests there was actually a positive value. In any case, it was always a very small difference

**Pull request merged**